# Data Setup

In [1]:
import pandas as pd

file_path = '../data/JSE_top40_2003_2018.csv' 
df = pd.read_csv(file_path)


## Data exploration

In [2]:
print(df.head())
print(df.info())

         date       ACL       ABG       ABI       AGL       AMS       ANG  \
0  2003-01-03  1.013216  1.002003  0.996377  1.009552  1.019327  1.000000   
1  2003-01-06  0.957608  1.010060  1.001091  0.993235  1.019878  1.000000   
2  2003-01-07  1.032690  1.007613  1.007991  1.020418  1.064468  1.009221   
3  2003-01-08  1.009013  0.994608  0.965766  0.998662  0.964507  0.991066   
4  2003-01-09  0.973856  1.022755  0.999067  0.975977  0.990070  1.000000   

        AOD       BAW       BIL  ...       REM       RMH       SAB       SAP  \
0  1.008197  1.017779  1.020143  ...  1.018547  1.018182  0.999168  0.998079   
1  1.008130  0.996723  1.051695  ...  0.996161  0.987500  1.008993  0.992308   
2  1.008065  1.018622  0.958098  ...  1.010587  1.000904  1.023436  1.013566   
3  1.000000  0.999781  0.992514  ...  1.004763  0.993677  0.987744  0.982793   
4  1.045600  1.010972  1.000848  ...  0.990519  1.004546  0.991020  1.011025   

        SBK       SLM       SNH       SOL       TBS     

# JSE Top40 Tickers (2003–2018)



| Heading | Definition / Company Name |
|---------|----------------------------|
| `date` | The trading date of the observation. |
| `ACL` | ArcelorMittal South Africa Limited |
| `ABG` | Absa Group Limited (formerly Barclays Africa Group) |
| `ABI` | Amalgamated Beverage Industries (legacy SABMiller entity) |
| `AGL` | Anglo American plc |
| `AMS` | Anglo American Platinum Limited |
| `ANG` | AngloGold Ashanti Limited |
| `AOD` | Historical JSE Constituent |
| `BAW` | Barloworld Limited |
| `BIL` | BHP Billiton plc (now BHP Group) |
| `BVT` | Bidvest Group Limited |
| `CFR` | Compagnie Financière Richemont SA |
| `CRH` | Historical JSE Constituent (e.g., Coronation Holdings) |
| `CRN` | Historical JSE Constituent |
| `DDT` | Dimension Data Holdings plc (legacy) |
| `DRD` | DRDGOLD Limited |
| `EXX` | Exxaro Resources Limited |
| `FSR` | FirstRand Limited |
| `GFI` | Gold Fields Limited |
| `HAR` | Harmony Gold Mining Company Limited |
| `IMP` | Impala Platinum Holdings Limited |
| `INL` | Investec Limited (South African listing) |
| `INP` | Investec plc (UK dual listing) |
| `IPL` | Imperial Holdings Limited (legacy) |
| `ITU` | Italtile Limited |
| `JNC` | Johnnic Holdings Limited (legacy) |
| `LGL` | Liberty Group Limited (legacy) |
| `MTN` | MTN Group Limited |
| `NED` | Nedbank Group Limited |
| `NPK` | Nampak Limited |
| `OML` | Old Mutual Limited |
| `PIK` | Pick n Pay Stores Limited |
| `REM` | Remgro Limited |
| `RMH` | RMB Holdings Limited |
| `SAB` | SABMiller plc (legacy) |
| `SAP` | Sappi Limited |
| `SBK` | Standard Bank Group Limited |
| `SLM` | Sanlam Limited |
| `SNH` | Steinhoff International Holdings N.V. (legacy) |
| `SOL` | Sasol Limited |
| `TBS` | Tiger Brands Limited |
| `VNF` | VenFin Limited (legacy) |

# Data Interpretation

Dataset is price relatives ($PR_t$).
A value greater than 1 (e.g., 1.013216 for ACL on 2003-01-03) means the stock price increased by approximately 1.32% compared to the previous trading day.
Similarly, a value less than 1 (e.g., 0.957608 for ACL on 2003-01-06) means the stock price decreased by approximately 4.24%.
Raw asset prices for JSE Top40 constituents would trade at wildly different magnitudes. This is more practical